generator/src/main/java/com/scottlogic/deg/generator/guice/FormatterProvider.java

```diff
@@ -0,0 +1,34 @@
+package com.scottlogic.deg.generator.guice;
+
+import com.google.inject.Inject;
+import com.google.inject.Provider;
+import com.scottlogic.deg.generator.generation.GenerationConfigSource;
+import com.scottlogic.deg.generator.outputs.datasetwriters.CsvFormatter;
+import com.scottlogic.deg.generator.outputs.datasetwriters.JsonFormatter;
+import com.scottlogic.deg.generator.outputs.datasetwriters.RowOutputFormatter;
+
+public class FormatterProvider implements Provider<RowOutputFormatter> {
+
+    private final GenerationConfigSource configSource;
+    private final CsvFormatter csvFormatter;
+    private final JsonFormatter jsonFormatter;
+
+    @Inject
+    public FormatterProvider(GenerationConfigSource configSource, CsvFormatter csvFormatter, JsonFormatter jsonFormatter) {
+        this.configSource = configSource;
+        this.csvFormatter = csvFormatter;
+        this.jsonFormatter = jsonFormatter;
+    }
+
+    @Override
+    public RowOutputFormatter get() {
```

Comment:
Always leave a default case on your switch, even if it throws an exception. This means that if someone later adds another value to the enum, you get an obvious error or logging, instead of silent failure.

Reply:
There is an explicit throw after the switch, which is equivalent to the default case?

Reply:
You're correct, I missed that.

I would consider inside the switch to be clearer to make it explicit that failure to find an option from the enum is the intent, but it's a minor point. Feel free to bypass this and mark it resolved.